api/prisma/reindex.ts

```diff
@@ -15,30 +15,48 @@ const reindex = async () => {
 
     const pubs = await client.prisma.publication.findMany({
         where: {
-            currentStatus: 'LIVE'
+            versions: {
+                some: {
+                    currentStatus: 'LIVE'
+                }
+            }
+        },
+        include: {
+            versions: {
+                where: {
+                    currentStatus: 'LIVE'
+                }
+            }
         }
     });
 
     console.log(`reindexing ${pubs.length}`);
 
     for (const pub of pubs) {
-        await client.search.create({
-            index: 'publications',
-            id: pub.id,
-            body: {
-                id: pub.id,
-                type: pub.type,
-                title: pub.title,
-                licence: pub.licence,
-                description: pub.description,
-                keywords: pub.keywords,
-                content: pub.content,
-                language: 'en',
-                currentStatus: pub.currentStatus,
-                publishedDate: pub.publishedDate,
-                cleanContent: htmlToText.convert(pub.content)
-            }
+        const latestLiveVersion = pub.versions.find((version) => {
```

Comment:
We can do this filtering into the prisma query directly to only include the last LIVE version of each publication. While running this script the search functionality is down. We want to keep it running as fast as possible.

Reply:
I can't think how to do it in one query - if there was a separate query for each publication you could do something like
```
client.prisma.publicationVersions.findFirst({
    where: {
        versionOf: publicationId,
        currentStatus: 'LIVE',
    },
    orderBy: {
        versionNumber: 'desc'
    }
});
```
But because we can't limit the "include" to only return one result like we can with a findFirst, I don't know how we can do it in the single query above.

Reply:
Actually, we can order the "included" versions by versionNumber descending and then access the latest one at versions[0]. I will do that